Sources/OktaIdx/IDXFormField.swift

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright (c) 2021, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+
+extension IDXClient.Remediation.Form {
+    /// Describes an individual field within a form, used to collect and submit information from the user to proceed through the authentication workflow.
+    ///
+    /// Nested form values can be accessed through keyed subscripting, for example:
+    ///
+    ///    credentials.form["passcode"]
+    @objc(IDXRemediationFormField)
+    final public class Field: NSObject {
+        /// The programmatic name for this form value.
+        @objc public let name: String?
+        
+        /// The user-readable label describing this form value.
+        @objc public let label: String?
+        
+        /// The type of value expected from the client.
+        @objc public let type: String?
+        
+        /// The value to send, if a default is provided from the Identity Engine.
+        @objc public var value: AnyObject? {
+            get { _value }
+            set {
+                guard isMutable else { return }
```

Comment:
It can be unexpected for a developer. Debugging may be hard. I suggest putting some logging here or assertion.

Reply:
Unfortunately we can't mark property getters/setters as `throws`. I agree that logging will be useful, though I'd probably leave that to a more comprehensive story to add logging end-to-end.